### PR TITLE
Improve testing of new functions

### PR DESCRIPTION
**Pull request by judofyr**
- `variables` must be an object, not an array
- Add more names to tests
- Test non-number elements in array::join

### CODE REVIEW

File: test/function/string-split.yml [hunk 1]
```diff
@@ -7,48 +7,44 @@ tests:
...
   variables:
-  - str: '["a", "b", "c"]'
-  - str: 'null'
-  - str: '"x"'
-  - str: '42'
+    val:
+    - '["a", "b", "c"]'
+    - '{}'
+    - 'null'
+    - '42'
```
Comment (atombender) [inline test/function/string-split.yml:49]: This should include a string on either side, which is why the variables and tests were split.

File: test/function/array-join.yml [hunk 2]
```diff
@@ -7,37 +7,49 @@ tests:
...
-    array::join(~arr~, ~sep~)
+    array::join(~arr~, "")
+  variables:
+    arr:
+    - 'null'
+    - '"x"'
+    - '42'
+  result: null
+
+- name: Invalid seprator types
```
Comment (atombender) [inline test/function/array-join.yml:46]:
```suggestion
- name: Invalid separator types
```

File: test/function/array-join.yml [hunk 3]
```diff
@@ -7,37 +7,49 @@ tests:
...
-  # Basic
-  - variables:
+  - name: Strings
+    variables:
       arr: '["a", "b", "c"]'
       sep: '"-"'
     result: "a-b-c"
-  # Empty separator
-  - variables:
+  - name: Empty seperator
```
Comment (atombender) [inline test/function/array-join.yml:15]:
```suggestion
  - name: Empty separator
```

**Review by atombender — CHANGES_REQUESTED** [review]

File: test/function/string-split.yml (hunk 1 above)
Comment (judofyr) [reply test/function/string-split.yml]: Better now?

**Review by atombender — APPROVED** [review]

**Pull request merged**